### PR TITLE
bitnami/kubeapps Improved handling of packaging options for Kubeapps

### PR DESCRIPTION
**Pull request by absoludity**
**Description of the change**

This change contains the changes landed in our Kubeapps chart at https://github.com/kubeapps/kubeapps/pull/4265 , but without the update to the plugin paths for now.

Adding Antonio and Rafa as reviewers just to ensure we're on the same page about landing this change now.

**Benefits**

It gives us a better UX for choosing which packaging system to use with Kubeapps, by allowing the user to choose just the packaging system, while the chart then calculates the required plugins. It also allows the chart to conditionally enable/disable chart dependencies based on the chosen packaging.

Backwards compatible as setting the `kubeappsapis.enabledPlugins` will override (see paste below)
**Possible drawbacks**

I understand the readme-generator-for-helm check failure is known (it doesn't deal with nil values yet), but I'm confused about the ```Error: Detected changes in charts without version bump in Chart.yaml.
Charts changed: 1
bitnami/kubeapps
Version bumps detected: 99```
error, given that I have bumped the patch version of the chart?

**Applicable issues**

  - fixes #

**Additional information**

Here's the output from testing the change by comparing the output of helm template:

```
# Default plugins are helm and resources:
➜  kubeapps git:(3695-dont-require-explicit-enable-redis-2) ✗ git --no-pager diff docs/user/manifests/kubeapps-local-dev-values.yaml
➜  kubeapps git:(3695-dont-require-explicit-enable-redis-2) ✗ make deploy-dev-kubeapps | grep "\- /kubeapps-apis" -A 5              
            - /kubeapps-apis
          args:
            - --plugin-dir
            - /plugins/helm
            - --plugin-dir
            - /plugins/resources

# Enabling carvel updates the enabled plugins.
➜  kubeapps git:(3695-dont-require-explicit-enable-redis-2) ✗ git --no-pager diff docs/user/manifests/kubeapps-local-dev-values.yaml
diff --git a/docs/user/manifests/kubeapps-local-dev-values.yaml b/docs/user/manifests/kubeapps-local-dev-values.yaml
index 3b7518cc..ff73074b 100644
--- a/docs/user/manifests/kubeapps-local-dev-values.yaml
+++ b/docs/user/manifests/kubeapps-local-dev-values.yaml
@@ -30,3 +30,6 @@ ingress:
     nginx.ingress.kubernetes.io/proxy-read-timeout: "600"
     # Required for ingress-nginx 1.0.0 for a valid ingress.
     kubernetes.io/ingress.class: nginx
+packaging:
+  carvel:
+    enabled: true
➜  kubeapps git:(3695-dont-require-explicit-enable-redis-2) ✗ make deploy-dev-kubeapps | grep "\- /kubeapps-apis" -A 7
            - /kubeapps-apis
          args:
            - --plugin-dir
            - /plugins/kapp-controller
            - --plugin-dir
            - /plugins/helm
            - --plugin-dir
            - /plugins/resources

# Enabling flux results in an error as helm is also enabled:
➜  kubeapps git:(3695-dont-require-explicit-enable-redis-2) ✗ git --no-pager diff docs/user/manifests/kubeapps-local-dev-values.yaml
diff --git a/docs/user/manifests/kubeapps-local-dev-values.yaml b/docs/user/manifests/kubeapps-local-dev-values.yaml
index 3b7518cc..7117cbaa 100644
--- a/docs/user/manifests/kubeapps-local-dev-values.yaml
+++ b/docs/user/manifests/kubeapps-local-dev-values.yaml
@@ -30,3 +30,8 @@ ingress:
     nginx.ingress.kubernetes.io/proxy-read-timeout: "600"
     # Required for ingress-nginx 1.0.0 for a valid ingress.
     kubernetes.io/ingress.class: nginx
+packaging:
+  carvel:
+    enabled: true
+  flux:
+    enabled: true
➜  kubeapps git:(3695-dont-require-explicit-enable-redis-2) ✗ make deploy-dev-kubeapps | grep "\- /kubeapps-apis" -A 7              
Error: execution error at (kubeapps/templates/kubeappsapis/deployment.yaml:71:35): packaging: Please enable only one of the flux and helm plugins, since they both operate on Helm releases.

Use --debug flag to render out invalid YAML
make: *** [script/makefiles/deploy-dev.mk:40: deploy-dev-kubeapps] Error 1

# Disabling helm gets things going as expected:
➜  kubeapps git:(3695-dont-require-explicit-enable-redis-2) ✗ git --no-pager diff docs/user/manifests/kubeapps-local-dev-values.yaml
diff --git a/docs/user/manifests/kubeapps-local-dev-values.yaml b/docs/user/manifests/kubeapps-local-dev-values.yaml
index 3b7518cc..ba928fcc 100644
--- a/docs/user/manifests/kubeapps-local-dev-values.yaml
+++ b/docs/user/manifests/kubeapps-local-dev-values.yaml
@@ -30,3 +30,10 @@ ingress:
     nginx.ingress.kubernetes.io/proxy-read-timeout: "600"
     # Required for ingress-nginx 1.0.0 for a valid ingress.
     kubernetes.io/ingress.class: nginx
+packaging:
+  carvel:
+    enabled: true
+  flux:
+    enabled: true
+  helm:
+    enabled: false
➜  kubeapps git:(3695-dont-require-explicit-enable-redis-2) ✗ make deploy-dev-kubeapps | grep "\- /kubeapps-apis" -A 7              
            - /kubeapps-apis
          args:
            - --plugin-dir
            - /plugins/kapp-controller
            - --plugin-dir
            - /plugins/fluxv2
            - --plugin-dir
            - /plugins/resources

# Adding some other key to the packaging value causes a fail:
Use --debug flag to render out invalid YAML
make: *** [script/makefiles/deploy-dev.mk:40: deploy-dev-kubeapps] Error 1
➜  kubeapps git:(3695-dont-require-explicit-enable-redis-2) ✗ git --no-pager diff docs/user/manifests/kubeapps-local-dev-values.yaml
diff --git a/docs/user/manifests/kubeapps-local-dev-values.yaml b/docs/user/manifests/kubeapps-local-dev-values.yaml
index 3b7518cc..c3a43886 100644
--- a/docs/user/manifests/kubeapps-local-dev-values.yaml
+++ b/docs/user/manifests/kubeapps-local-dev-values.yaml
@@ -30,3 +30,12 @@ ingress:
     nginx.ingress.kubernetes.io/proxy-read-timeout: "600"
     # Required for ingress-nginx 1.0.0 for a valid ingress.
     kubernetes.io/ingress.class: nginx
+packaging:
+  carvel:
+    enabled: true
+  flux:
+    enabled: true
+  helm:
+    enabled: false
+  unknownpkg:
+    enabled: true
➜  kubeapps git:(3695-dont-require-explicit-enable-redis-2) ✗ make deploy-dev-kubeapps | grep "\- /kubeapps-apis" -A 7              
Error: execution error at (kubeapps/templates/kubeappsapis/deployment.yaml:71:35): packaging: Unsupported packaging option: unknownpkg

Use --debug flag to render out invalid YAML
make: *** [script/makefiles/deploy-dev.mk:40: deploy-dev-kubeapps] Error 1

# Manually overriding the enabled plugins is still possible:
➜  kubeapps git:(3695-dont-require-explicit-enable-redis-2) ✗ git --no-pager diff docs/user/manifests/kubeapps-local-dev-values.yaml
diff --git a/docs/user/manifests/kubeapps-local-dev-values.yaml b/docs/user/manifests/kubeapps-local-dev-values.yaml
index 3b7518cc..e1fe4b91 100644
--- a/docs/user/manifests/kubeapps-local-dev-values.yaml
+++ b/docs/user/manifests/kubeapps-local-dev-values.yaml
@@ -18,6 +18,9 @@ postgresql:
   existingSecret: postgresql-db
 kubeappsapis:
   replicaCount: 1
+  enabledPlugins:
+  - unknownpkg
+  - anotherunknownpkg
 ingress:
   enabled: true
   hostname: localhost
@@ -30,3 +33,12 @@ ingress:
     nginx.ingress.kubernetes.io/proxy-read-timeout: "600"
     # Required for ingress-nginx 1.0.0 for a valid ingress.
     kubernetes.io/ingress.class: nginx
+packaging:
+  carvel:
+    enabled: true
+  flux:
+    enabled: true
+  helm:
+    enabled: false
+  unknownpkg:
+    enabled: true
➜  kubeapps git:(3695-dont-require-explicit-enable-redis-2) ✗ make deploy-dev-kubeapps | grep "\- /kubeapps-apis" -A 7              
            - /kubeapps-apis
          args:
            - --plugin-dir
            - /plugins/unknownpkg
            - --plugin-dir
            - /plugins/anotherunknownpkg
            - --clusters-config-path=/config/clusters.conf
            - --plugin-config-path=/config/kubeapps-apis/plugins.conf
```

**Checklist**

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the values.yaml and added to the README.md using (readme-generator-for-helm)[https://github.com/bitnami-labs/readme-generator-for-helm]
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/kubeapps/Chart.yaml [hunk 1]
```diff
@@ -33,4 +33,4 @@ maintainers:
 name: kubeapps
 sources:
   - https://github.com/kubeapps/kubeapps
-version: 7.8.0
+version: 7.8.1
```
Comment (castelblanque) [inline bitnami/kubeapps/Chart.yaml:36]: Shouldn't we add the `-dev1` suffix?

File: bitnami/kubeapps/README.md [hunk 2]
```diff
@@ -492,7 +501,7 @@ Once you have installed Kubeapps follow the [Getting Started Guide](https://gith
 
 | Name                                                                                            | Description                                                                               | Value                   |
 | ----------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------- | ----------------------- |
-| `kubeappsapis.enabledPlugins`                                                                   | Enabled plugins for the Kubeapps-APIs service                                             | `["helm","resources"]`  |
+| `kubeappsapis.enabledPlugins`                                                                   | Manually override which plugins are enabled for the Kubeapps-APIs service                 | `nil`                   |
```
[the remark this answers is not in the document]
Comment (castelblanque) [reply bitnami/kubeapps/README.md:504]: I mean for the readme, I see it mentioned in the full comment of values.yaml.

**Comment by castelblanque** [comment]
LGTM!

Sorry for not pointing this out in the original PR on Kubeapps, if this is set up:
```
packaging:
  [...]
  flux:
    enabled: true
[...]
kubeappsapis:
  enabledPlugins:
    - helm
    - resources
```
Installation ends up having Redis but no Flux. I know it might be an edge case, but to avoid this kind of duplicity why not removing the `enabledPlugins` option completely?

**Comment by absoludity** [comment]
> LGTM!
> 
> Sorry for not pointing this out in the original PR on Kubeapps, if this is set up:
> 
> ```
> packaging:
>   [...]
>   flux:
>     enabled: true
> [...]
> kubeappsapis:
>   enabledPlugins:
>     - helm
>     - resources
> ```
> 
> Installation ends up having Redis but no Flux. I know it might be an edge case, but to avoid this kind of duplicity why not removing the `enabledPlugins` option completely?

As Antonio pointed out elsewhere, if people create their own plugins, or we work on new plugins, we may want to set the enabledPlugins explicitly.

**Comment by absoludity** [comment]
Merged master and fixed the issue I mentioned earlier (changes from https://github.com/kubeapps/kubeapps/pull/4309 )

**Review by antgamdia — APPROVED** [review]
Great! Thanks for the changes, much better UX for enabling the packaging plugins.

**Comment by javsalgar** [comment]
Could you check the issues in the actions? It seems that there's missing metadata in the README

**Comment by antgamdia** [comment]
> Could you check the issues in the actions? It seems that there's missing metadata in the README

It's a known issue in the readme-generator project, https://github.com/bitnami-labs/readme-generator-for-helm/issues/31

**Comment by absoludity** [comment]
> > Could you check the issues in the actions? It seems that there's missing metadata in the README
> 
> It's a known issue in the readme-generator project, [bitnami-labs/readme-generator-for-helm#31](https://github.com/bitnami-labs/readme-generator-for-helm/issues/31)

Can we merge this, given that the issue for the failing check is known. We don't need to wait for that issue in the readme generator to be fixed to land this (as we similarly landed #8981)

**Pull request merged**